browser/android/preferences/website/desktop_mode_tab_helper.cc

```diff
@@ -46,4 +46,17 @@ void DesktopModeTabHelper::NavigationEntryCommitted(
       UpdateOverridingUserAgent();
 }
 
+void DesktopModeTabHelper::DidStartNavigation(
+    content::NavigationHandle* navigation_handle) {
+  // We need to override UA on the first navigation only.
+  if (!navigation_started_) {
+    navigation_started_ = true;
+    need_override_ua_ = true;
+    return;
+  }
+
+  // We should not override UA on all the next navigations.
+  need_override_ua_ = false;
```

Comment:
Hmm, I think this PR's changes seems same with initial version as first navigation doesn't have last committed entry?
```
  content::NavigationEntry* entry =
      web_contents()->GetController().GetLastCommittedEntry();
  need_override_ua_ = entry == nullptr;
```

Reply:
Somehow it does, with `need_override_ua_ = entry == nullptr` `need_override_ua_` is always `false` in `cr98`